lms/djangoapps/api_manager/sessions/views.py

```diff
@@ -105,21 +111,29 @@ def post(self, request):
                     # violate our RESTfulness
                     #
                     engine = import_module(settings.SESSION_ENGINE)
-                    new_session = engine.SessionStore()
-                    new_session.create()
+                    if session_id is None:
```

Comment:
You might consider swapping the if/else conditions based on what you feel the primary use case will be -- requests having a session_id or not